hbase_storage/src/test/java/com/splicemachine/si/impl/TimeTravelTest.java

```diff
@@ -0,0 +1,137 @@
+package com.splicemachine.si.impl;
+
+import com.splicemachine.primitives.Bytes;
+import com.splicemachine.si.constants.SIConstants;
+import com.splicemachine.si.impl.region.TxnFinder;
+import com.splicemachine.si.impl.region.TxnTimeTraveler;
+import com.splicemachine.utils.Pair;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.sql.Timestamp;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+public class TimeTravelTest {
+
+    private static Long toTs(String s) throws ParseException {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss.SSS");
+        Date parsedDate = dateFormat.parse(s);
+        Timestamp timestamp = new java.sql.Timestamp(parsedDate.getTime());
+        return timestamp.getTime();
+    }
+
+    class MockTxnFinder implements TxnFinder {
+
+
+        List<SortedMap<Long, Long>> txnTimeline = new ArrayList<>();
+
+        MockTxnFinder() {
+            for(int i = 0; i < SIConstants.TRANSACTION_TABLE_BUCKET_COUNT; ++i) {
+                txnTimeline.add(new TreeMap<>());
+            }
+        }
+
+        MockTxnFinder AddTxn(int bucket, long txnId, String ts) throws ParseException {
+            assert bucket >= 0 && bucket < SIConstants.TRANSACTION_TABLE_BUCKET_COUNT;
+            txnTimeline.get(bucket).put(txnId, toTs(ts));
+            return this;
+        }
+
+        @Override
+        public Pair<Long, Long> find(byte bucket, byte[] begin, boolean reverse) throws IOException {
+            if(begin == null) {
```

Comment:
`SortedMap<Long, Long> m = txnTimeline.get(bucket);` and `if(m.size() == 0) { return null`
can be done independent of if(begin == null) and if(reverse).

I think you can add at the beginning of this function
```
SortedMap<Long, Long> m = txnTimeline.get(bucket);
if(m.size() == 0) {
    return null;
}
```
and remove in other lines.